### ISSUE TITLE
Zombie Barriers (Reference: Call Of Duty Zombies) 

### ISSUE THREAD

**Issue by AskJayden** [garbagemule/MobArena#668]
Been using Mob Arena for 6 years. 

Eventually I created a map similar to that of Call Of Duty World At War. I config the plugin so that only zombies spawn and they come in waves. Something that bothers me a little bit is that I have to make my spawnpoints in the middle of the map or they are set up so that the zombies spawn in front of the player when they are playing. I would love to see barriers being implemented into the plugin. Obviously without mods I can see it being impossible. 

Basically I was thinking I could use a door, a non-gated fence or a slab or something to be the ' Barrier' and that zombies can break the block. So I'm thinking it could be cool to see doors being broken constantly and consistently smashed by Zombies.

**Comment by garbagemule** [comment]
There is an event in the Bukkit API related to zombies breaking doors: [EntityBreakDoorEvent](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/EntityBreakDoorEvent.html)

Unfortunately, the Bukkit API exposes no methods to set the door-breaking capability of a zombie. It looks like it's an [NBT](https://www.digminecraft.com/data_tags/zombie.php), which requires a dependency on a server implementation or reflection, so I don't see the "consistency" aspect of your proposal being added to MobArena any time soon.

If the doors are placed by players during the session, they should get cleaned up afterwards if they haven't been broken yet. So technically, if you can find a way to force the NBT on all zombies in the region via other means, I think all the pieces of the puzzle are already there.

**Comment by Ghmmy** [comment]
You might be able to achieve this using mythicmobs that spawn special zombies that can break doors. Either by NBT or a special skill that removes a door block after a while.

Another thing, if you _really_ want to make a remake of COD zombies, there are/is a plugin out there that does exactly what you want. It takes a while to set up all the spawnpoints, unlockable areas, custom items, etc, but it will be worth it once its all working. 

If you want to achieve this through only mobarena and have never messed around with MythicMobs, it could take a while to get the required experienced needed to make it all working with just MA and MM, hence why I suggested the alternative plugin for the gamemode you are working on.

If, however, you are well-experienced with mythicmobs, it shouldnt be a big problem setting those things up either way.

**Comment by AskJayden** [comment]
Thanks for reaching out to me.

I've dabbled in your plugin and made a COD zombies experienced before.Once
I get it set up would love to have you on the server to check it out and
give feedback. Might even spark ideas for updates !

**Comment by Ghmmy** [comment]
I'm not someone who joins servers and write reviews or anything, as my taste is entirely different and I'm rather critical when it comes to configuration. You do you and as long as you in particulair are satisfied with the outcome.
